### DISCUSSION TITLE
Atividades cap 2

### DISCUSSION THREAD

**Discussion by vitorestevam** [clube-de-leitura-python/pense-em-python-2021#11] [Perguntas & Respostas]
Essa discussão foi uma sugestão que vi em alguma outra conversa. Aqui a gente pode botar como resolveu as atividades e tirar dúvidas 🚀

**Comment by vitorestevam** [comment]
## Questão 2.3 😢 
Como vocês fizeram a questão que fala sobre a corrida e os tempos?  Tive um pouco de dificuldade e acabei desistindo de converter o tempo manualmente e apelei pra biblioteca ``` datetime ``` que permite operar diretamente as horas, mas acho que intuito dela era fazer tudo na matemática msm.

**Comment by villares** [comment]
Fiz assim ó, veja se ajuda
```python
tempo_primeira_corrida = 8 * 60 + 15   # 8 minutos e 15 segundos em segundos

velocidade2_km_por_s =  1 / (7 * 60 + 12)   #  não sei se precisava calcular a velocidade...
```

**Reply by vitorestevam** [reply]
Essa parte de transformar de hora pra seg eu até consegui fazer
``` python 
hora_inicio = ((6*60)+52)*60 #converte de hora pra minutos, então converte pra segundos
tempo_corrida_1 = 1 * (8*60)+15 #converte de minutos pra segundos
tempo_corrida_2 = 3 * (7*60)+12 #converte de minutos pra segundos
tempo_corrida_3 = 1 * (8*60)+15 #converte de minutos pra segundos

tempo_total = tempo_corrida_1 + tempo_corrida_2 + tempo_corrida_3 #em segundos
hora_final = hora_inicio + tempo_total #em segundos

#falta reconverter pra horas e minutos
```
Eu me compliquei foi depois pra transformar de segundo de volta pra hora. Dei uma pesquisada e acho que pra reconverter isso  dá pra eu usar modulo e floor

https://blog.teclado.com/pythons-modulo-operator-and-floor-division/

**Reply by villares** [reply]
é essa parte de converter de volta é meio chata.
Talvez fazer assim ó, confere pra mim...
```python
# hora_final é um número de segundos
horas_horario_final = hora_final // 360  # horas_hora_final é em horas,  // floor division, joga fora minutos.
minutos_horario_final = int(hora_final / 60 - horas_hora_final * 60)  # em minutos, truncado para um numero inteiro
```

**Reply by vitorestevam** [reply]
Deu bom! Me baseei no teu e nesse link que mandei aí e fiz assim

``` python
segundos = hora_final_em_segundos
hora_final_s =  segundos % 60
minutos = segundos // 60
hora_final_m = minutos % 60
horas = minutos // 60
hora_final_h = horas
print(f"{hora_final_h}:{hora_final_m}:{hora_final_s}") #7:29:42
```

vlw 😃
